src/components/FilterSettings/FilterSettings.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { FC, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { Alert, Button, ModalFooter } from '@keen.io/ui-core';
+
+import {
+  BoldMessage,
+  Content,
+  Container,
+  ConnectionsContainer,
+  CancelButton,
+  Description,
+  DetachedConnections,
+  DetachedConnectionItem,
+  EmptyConnections,
+  ErrorContainer,
+  FieldGroup,
+  Field,
+  FooterContent,
+  NormalMessage,
+} from './FilterSettings.styles';
+
+import { EventStream, TargetProperty, TooltipHint } from './components';
+import WidgetConnections from '../WidgetConnections';
+
+import {
+  applySettings,
+  setEventStream,
+  setTargetProperty,
+  getFilterSettings,
+  updateConnection,
+} from '../../modules/filter';
+import { getCurrentDashboardChartsCount } from '../../modules/dashboards';
+
+import { ERRORS } from './constants';
+
+import { FilterSettingsError } from './types';
+
+type Props = {
+  /* Cancel filter settings */
+  onCancel: () => void;
+};
+
+const FilterSettings: FC<Props> = ({ onCancel }) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const [error, setError] = useState<FilterSettingsError>(null);
+
+  const {
+    widgetConnections,
+    detachedWidgetConnections,
+    eventStream,
+    targetProperty,
+    eventStreamsPool,
+    eventStreamSchema,
+    schemaProcessing,
+  } = useSelector(getFilterSettings);
+  const chartWidgetsCount = useSelector(getCurrentDashboardChartsCount);
+
+  const [isEditMode] = useState(!!targetProperty);
```

Comment:
do we need `useState` for this?

Reply:
+1

Reply:
Yes, as `targetProperty` could change in store and the button label should stay the same. It is used to show `create` or `edit` label.